tests/rptest/scale_tests/many_topics_test.py

```diff
@@ -0,0 +1,1296 @@
+
+        # Select random topic from the list
+        target_topic = topic_names[random.randint(0, len(topic_names))]
+        # Consumer specific config
+        consumer_extra_config = {
+            "auto.offset.reset": "smallest",
+            "group.id": "topic_swarm_group"
+        }
+        self.logger.info(f"Start consuming from {target_topic}")
+        # Consumer
+        start_time_s = time.time()
+        consumer = kclient.get_consumer(consumer_extra_config)
+        numbers = []
+        elapsed = 0.0
+        # Message consuming loop
+        try:
+            consumer.subscribe([target_topic])
+            while True:
+                # calculate elapsed time
+                elapsed = time.time() - start_time_s
+
+                # Exit on target number reached or timeout
+                if len(numbers) == message_count or elapsed > timeout_sec:
+                    break
+
+                # Poll for the message
+                msg = consumer.poll(timeout=1.0)
+                if msg is None:
+                    continue
+                # On error, check for the EOF
+                if msg.error():
+                    if msg.error().code() == KafkaError._PARTITION_EOF:
+                        # End of partition event
+                        self.logger.info(f"Consumer of '{msg.topic()}' "
+                                         f"[{msg.partition()}] reached "
+                                         f"end at offset {msg.offset()}")
+                        break
+                    # If not EOF, raise it
+                    elif msg.error():
+                        raise KafkaException(msg.error())
+                else:
+                    # Save value from the message
+                    numbers.append(int(msg.value()))
+        finally:
+            # Close down consumer to commit final offsets.
+            consumer.close()
+
+        self.logger.info(f"Consumed {len(numbers)} messages")
+        # Check that we received all numbers
+        if elapsed > timeout_sec:
+            raise RuntimeError("Timeout consuming messages "
+                               f"from {target_topic}")
+        elif not check_consecutive(numbers):
+            raise RuntimeError("Produced and consumed messages mismatch "
+                               f"for {target_topic}")
+
+        return
+
+    def _write_and_random_read_many_topics(self, message_count, num_topics,
+                                           topic_names):
+        """
+            Test checks that each of the many topics can be written to.
+            This produce/consume implementation will check actual data
+            of the messages to ensure that all of the messages are delivered
+
+            Pick X number of topics, write 100 messages in each
+            Pick random one among them, consume all messages
+            Iterate.
+        """
+        # Prepare librdkafka python client
+        kclient = PythonLibrdkafka(self.redpanda)
```

Comment:
As I mentioned in the past I don't think this is great because this will run on the ducktape monitoring/client node.

Reply:
Yes, and I had an idea to implement it inside remote script additions. But that is really a lot of effort at this point and we are already really tight on schedule. So this is merely a minimal load generated for the topic creation script.